operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/AbstractDependentResource.java

```diff
@@ -33,9 +33,11 @@ public AbstractDependentResource() {
       if (maybeActual.isEmpty()) {
         if (creatable) {
           var desired = desired(primary, context);
-          logForOperation("Creating", primary, desired);
-          var createdResource = handleCreate(desired, primary, context);
-          return ReconcileResult.resourceCreated(createdResource);
+          if (desired != null) {
```

Comment:
TBH, my opinion is we should not try to solve this case, we should simply throw exception in the desired as it is now. We are hacking the situation what should not happen

Reply:
> We are hacking the situation what should not happen

I agree that in the case of this sample, this is pretty unlikely, but this is definitely a situation that can happen for others inter-dependent use case.
I understand this is not yet supported by dependent for now and if required, it could be handle with a standalone implementation that would catch the exception for eg or throw a a specialized exception to be handle in the errorHandler so it can Update the primary status to say that the first dependent is ok but not the second one. Would be a bit tricky but can de done.

Should it be described in the doc that throwing an exception is a valid "return" for desired and create ?

Reply:
Relying on exceptions for flow control is a bad pattern but that's what we need to do until we implement better flow control of dependents.

Reply:
> Relying on exceptions for flow control is a bad pattern but that's what we need to do until we implement better flow control of dependents.

agreed. closing this

Reply:
@metacosm @scrocquesel regarding the exception I created a separate issue to discuss, I also added my concerns regarding to enum extension. But more then happy to discuss it in depth under this issue: https://github.com/java-operator-sdk/java-operator-sdk/issues/1126